### PR TITLE
FS-3685-view-assessment-activity-history-for-an-application

### PR DESCRIPTION
**Pull request by ramsharma-prog**
As an assessor (QA User) I need to see the application activity log so that I can check the quality of my work

- [X] Unit tests and other appropriate tests added or updated
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
Unit tests added 

Screenshot 2023-11-24 at 19 46 26

Screenshot 2023-11-24 at 19 43 33

### CODE REVIEW

File: db/queries/assessment_records/queries.py [hunk 1]
```diff
@@ -661,50 +668,238 @@ def get_assessment_records_score_data_by_round_id(
...
+            TagAssociation.application_id == application_id,
+        ).all()
+        list_existing_tags = defaultdict(list)
+        for existing_tag in existing_tags:
+            tag_id = str(existing_tag.tag_id)
+            list_existing_tags[tag_id].append(
+                (existing_tag.created_at, existing_tag)
             )
-            db.session.add(new_tag)
+        return list_existing_tags
```
[the remark this answers is not in the document]
Comment (ramsharma-prog) [reply db/queries/assessment_records/queries.py:704]: No it doesn't I could move this out  to tidy it up 😄

File: db/queries/assessment_records/queries.py [hunk 2]
```diff
@@ -661,50 +668,238 @@ def get_assessment_records_score_data_by_round_id(
...
+                        ),
+                        None,
+                    )
+                    if _incoming_tag:
+                        current_app.logger.info(
+                            f"Tag_id is already associated: {existing_tag_id}"
+                        )
+                    else:
+                        filtered_tags.append(existing_tag_list)
+                return filtered_tags
```
[the remark this answers is not in the document]
Comment (ramsharma-prog) [reply db/queries/assessment_records/queries.py:779]: No it doesn't I could move this out to tidy it up 😄

File: db/queries/assessment_records/queries.py [hunk 3]
```diff
@@ -661,50 +668,238 @@ def get_assessment_records_score_data_by_round_id(
...
+                if most_recent_tag.associated:
+                    current_app.logger.info(
+                        f"Dis-associating existing associated tag_id: {most_recent_tag.tag_id}"
+                    )
+                    create_tag(
+                        application_id,
+                        most_recent_tag.tag_id,
+                        False,
+                        incoming_user_id,
+                    )
```
[the remark this answers is not in the document]
Comment (ramsharma-prog) [reply db/queries/assessment_records/queries.py:794]: Sure! 😄

**Review by tferns — APPROVED** [review]
thanks for making those changes!

**Pull request merged**